### PR TITLE
SONARIAC-457 CMD Basic implementation to support shell form

### PR DESCRIPTION
**Pull request by mstachniuk**

### CODE REVIEW

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/CmdTreeImplTest.java [hunk 1]
```diff
@@ -67,7 +86,7 @@ void shouldCheckParseCmdTree() {
   }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/CmdTreeImplTest.java:87]: You are missing test case for shell form, which would highlight a `NullPointerException` in `CmdTreeImpl` ;) 
```suggestion


  @Test
  void shouldCheckParseCmdShellFormTree() {
    CmdTree tree = DockerTestUtils.parse("CMD executable param1 param2", DockerLexicalGrammar.CMD);
    assertThat(tree.getKind()).isEqualTo(DockerTree.Kind.CMD);
    assertThat(tree.keyword().value()).isEqualTo("CMD");

    assertThat(tree.cmdArguments().stream().map(TextTree::value)).containsExactly("executable", "param1", "param2");
    assertThat(((SyntaxToken)tree.children().get(0)).value()).isEqualTo("CMD");

    ExecFormTree execForm = (ExecFormTree) tree.children().get(1);
    assertThat(execForm).isSameAs(tree.execForm());
  }
```

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java [hunk 2]
```diff
@@ -27,15 +27,18 @@
...
 
   private final ExecFormTree execForm;
+  private final ShellFormTree shellForm;
 
-  public CmdTreeImpl(SyntaxToken keyword, @Nullable ExecFormTree execForm) {
+  public CmdTreeImpl(SyntaxToken keyword, @Nullable ExecFormTree execForm, @Nullable ShellFormTree shellForm) {
     super(keyword);
     this.execForm = execForm;
+    this.shellForm = shellForm;
   }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java:42]: You forgot to take care of this new attribute `shellForm` in the method `cmdArguments` ;)
Also as mentioned in my other comment in `CmdTreeImplTest.java`, you don't have a test case about shellForm, so you didn't saw this miss + the resulting NullPointerExeption.

**Review by rudy-regazzoni-sonarsource — CHANGES_REQUESTED** [review]
Nice work, but you currently have an issue in your code, please see my comments.
I like the fact that the method `cmdArguments` will give the arguments irrespective of the form (shell or exec), so that we don't need to handle this on the rule side.
You can ping me back when the issue is fixed and I will do a second review promptly ;)

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java (hunk 2 above)
Comment (mstachniuk) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java:43]: Excellent catch!
Yes, I forgot to extend `cmdArguments()`. By the way, I also extended `CmdTree` to care `shellForm`.

File: iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/CmdTreeImplTest.java (hunk 1 above)
Comment (mstachniuk) [reply iac-extensions/docker/src/test/java/org/sonar/iac/docker/tree/impl/CmdTreeImplTest.java:90]: Thank you for the test suggestion

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java [hunk 3]
```diff
@@ -60,17 +61,32 @@ public Kind getKind() {
...
+        for (ExecFormLiteralTree element : literals.elements()) {
+          result.add(element.value());
+        }
+      }
+    }
+    if (shellForm != null) {
+      List<SyntaxToken> literals = shellForm.literals();
+      if (literals != null) {
+        result.addAll(literals);
       }
```
Comment (rudy-regazzoni-sonarsource) [inline iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java:90]: I would tend to skip the second check for null `literals != null` as the method always return an arrays and don't has a Nullable/CheckForNull annotation.
Same for execForm, WDYT ?

**Review by rudy-regazzoni-sonarsource — APPROVED** [review]
LGTM ! See my comment.

File: iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java (hunk 3 above)
Comment (mstachniuk) [reply iac-extensions/docker/src/main/java/org/sonar/iac/docker/tree/impl/CmdTreeImpl.java:90]: ok

**Pull request merged**